book/workflow.rst

```diff
@@ -0,0 +1,236 @@
+    workflows:
+        phone_call:
+            label: 'Call Customer'
+            entity: Acme\Bundle\DemoBundle\Entity\PhoneCall
+            enabled: true
+            start_step: start_call
+
+The basic configuration contains a name that is displayed in the UI, the entity the workflow will
+be applied to and the initial step that is applied when initializing the workflow for an entity.
+
+Defining the Workflow Steps
+...........................
+
+Next, you need to define the states an entity can have when the workflow is applied:
+
+.. code-block:: yaml
+    :linenos:
+
+    workflows:
+        phone_call:
+            # ...
+            steps:
+                start_call:
+                    label: 'Start Phone Call'
+                    allowed_transitions:
+                        - connected
+                        - not_answered
+                start_conversation:
+                    label: 'Call Phone Conversation'
+                    allowed_transitions:
+                        - end_conversation
+                end_call:
+                    label: 'End Phone Call'
+                    is_final: true
+
+In this example, the entity can have three states: ``start_call`` (which was configured as the
+start step above), ``start_conversation`` and ``end_call``. The ``end_call`` step is marked as
+final which means that the workflow terminates when this step is reached.
+
+The ``allowed_transitions`` defines which :ref:`transitions <book-workflow-transitions>` can be
+applied in a step.
+
+Workflow Attributes
+...................
+
+During the transition from one step to another step, the user can modify attributes of the
+underlying entity. All attributes that can be modified in any of the transitions must be enabled
+with the ``attributes`` option:
+
+.. code-block:: yaml
+    :linenos:
+
+    workflows:
+        phone_call:
+            # ...
+            attributes:
+                phone_call:
+                    label: Phone Call
+                    type: entity
+                    options:
+                        class: Acme\Bundle\DemoWorkflowBundle\Entity\PhoneCall
+                call_timeout:
+                    type: integer
+                    label: 'Call Timeout'
+                call_successfull:
+                    type: boolean
+                    label: 'Call Successful'
+                conversation_successful:
+                    type: boolean
+                    label: 'Conversation Successful'
+                conversation_comment:
+                    type: string
+                    label: 'Conversation Comment'
+                conversation_result:
+                    type: string
+                    label: 'Conversation Result'
+                conversation:
+                    type: entity
+                    label: 'Conversation'
+                    options:
+                        class: Acme\Bundle\DemoWorkflowBundle\Entity\PhoneConversation
+
```

Comment:
Thanks, I have added a tip below: https://github.com/orocrm/documentation/pull/169/files#diff-3409be6f125c94b749d1d326d2bba0c4R121

Would be nice if you can confirm that the changes are right. I was not completely sure if I properly understood the use case for the `entity_attribute` option.